lib/std/paths.nim

```diff
@@ -0,0 +1,322 @@
+import std/private/osseps
+export osseps
+
+from std/private/ospaths2 {.all.} import joinPathImpl, joinPath, splitPath,
```

Comment:
Can be done later but `.all` here is not required. Instead export "all" in "private/ospaths2.nim" as it's a private module.

Reply:
Yeah, but "std/private/ospaths2" is imported and re-exported by `std/os`

Reply:
So? Only re-export in std/os what should be re-exported.

Reply:
done